frontend/src/components/grantaccess/GrantAccess.vue

```diff
@@ -0,0 +1,397 @@
+<script setup lang="ts">
+import { onMounted, ref } from 'vue';
+import router from '@/router';
+import { Form as VeeForm, Field, ErrorMessage } from 'vee-validate';
+import { number, object, string } from 'yup';
+
+import Button from 'primevue/button';
+import InputText from 'primevue/inputtext';
+import Dropdown from 'primevue/dropdown';
+import RadioButton from 'primevue/radiobutton';
+import ForestClientCard from './ForestClientCard.vue';
+
+import { ApiServiceFactory } from '@/services/ApiServiceFactory';
+import {
+    selectedApplication,
+    selectedApplicationDisplayText,
+} from '@/store/ApplicationState';
+import {
+    grantAccessFormData,
+    domainOptions,
+    FOREST_CLIENT_INPUT_MAX_LENGTH,
+    setGrantAccessFormData,
+    getGrantAccessFormData,
+} from '@/store/GrantAccessDataState';
+
+import type { FamApplicationRole, FamForestClient } from 'fam-api';
+
+const defaultFormData = {
+    domain: domainOptions.IDIR,
+    userId: '',
+    forestClientNumber: '',
+    role_id: null as number | null,
+};
+const formData = ref(defaultFormData); // clone default input
```

Comment:
this need to be a hard copy, just copy the value,  "ref(JSON.parse(JSON.stringify(defaultFormData)))", otherwise when you reset the form in the "cancelForm()" method, it won't get reset